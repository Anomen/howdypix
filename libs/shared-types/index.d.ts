export type MessageProcess = {
  root: string;
  path: string;
};
export type ExifData = {
  ISO?: number;
  make?: string;
  model?: string;
  createDate?: string;
};
