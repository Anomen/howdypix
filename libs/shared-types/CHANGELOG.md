# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

### Changed

- Simplify types for RabbitMQ messages ([#132](https://github.com/vickev/howdypix/pull/132))

## [0.1.0] - 2020-04-09

### Changed

- Version bump because lerna

## [0.0.2] - 2020-03-07

### Added

- Version bump because lerna


