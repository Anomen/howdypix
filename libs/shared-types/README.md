# @howdypix/shared-types

This package exposes common typescript types to be consumed by the HowdyPix applications. For example, it might be the data types sent to RabbitMQ, and consumed on the other side, adding the same type check on both client and server side.
