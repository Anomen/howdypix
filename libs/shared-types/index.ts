export type MessageProcess = {
  root: string;
  path: string;
};

export type ExifData = {
  ISO?: number;
  make?: string;
  model?: string;
  createDate?: string;
};

export type ProcessData = {
  exif: ExifData;
  thumbnails: string[];
};

export enum QueueName {
  TO_PROCESS = "toProcess",
  PROCESSED = "processed"
}
