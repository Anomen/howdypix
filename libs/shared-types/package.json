{
  "name": "@howdypix/shared-types",
  "version": "0.1.0",
  "license": "MIT",
  "main": "./build/index.js",
  "types": "./build/index.d.ts",
  "scripts": {
    "dev": "tsc --watch",
    "build": "rm -rf build && tsc",
    "lint": "eslint *.{ts,tsx}",
    "typecheck": "tsc --noEmit"
  },
  "devDependencies": {
    "@howdypix/eslint-config": "^0.1.0",
    "@howdypix/typescript-config": "^0.1.0"
  },
  "dependencies": {}
}
