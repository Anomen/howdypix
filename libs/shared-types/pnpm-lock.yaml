dependencies:
  typescript: registry.npmjs.org/typescript/3.6.3
devDependencies:
  '@howdypix/eslint-config': 'link:../eslint-config'
  '@howdypix/typescript-config': 'link:../typescript-config'
lockfileVersion: 5.1
packages:
  registry.npmjs.org/typescript/3.6.3:
    dev: false
    engines:
      node: '>=4.2.0'
    hasBin: true
    name: typescript
    resolution:
      integrity: sha512-N7bceJL1CtRQ2RiG0AQME13ksR7DiuQh/QehubYcghzv20tnh+MQnQIuJddTmsbqYj+dztchykemz0zFzlvdQw==
      registry: 'https://registry.npmjs.org/'
      tarball: 'https://registry.npmjs.org/typescript/-/typescript-3.6.3.tgz'
    version: 3.6.3
specifiers:
  '@howdypix/eslint-config': ^0.1.0
  '@howdypix/typescript-config': ^0.1.0
  typescript: ^3.6.3
