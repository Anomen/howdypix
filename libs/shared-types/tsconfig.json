{
  "extends": "@howdypix/typescript-config/base",
  "exclude": ["node_modules", "**/*.test.ts"],
  "include": ["*.ts"],
  "compilerOptions": {
    "declaration": true,
    "declarationDir": "./build",
    "outDir": "./build"
  }
}
