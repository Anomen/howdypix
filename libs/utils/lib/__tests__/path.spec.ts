import { omit } from "lodash";
import { HFile } from "@howdypix/shared-types";
import {
  hjoin,
  hparse,
  path2hfile,
  hfile2path,
  thumbnailPath,
  splitHowdyfiles,
} from "..";

describe("path", () => {
  const baseHfile: HFile = {
    dir: "dir",
    file: "file.jpg",
    source: "source",
    name: "name",
  };

  describe("hjoin", () => {
    it("should return the right value", () => {
      expect(hjoin(baseHfile)).toEqual("@source:dir/file.jpg");
    });

    it("should return the right value without dir", () => {
      expect(hjoin(omit(baseHfile, ["dir"]))).toEqual("@source:file.jpg");
    });

    it("should return the right value without file", () => {
      expect(hjoin(omit(baseHfile, ["file"]))).toEqual("@source:dir");
    });

    it("should return the right value without dir and file", () => {
      expect(hjoin(omit(baseHfile, ["dir", "file"]))).toEqual("@source:.");
    });
  });

  describe("hparse", () => {
    it("should return the right value", () => {
      expect(hparse("@source:dir/file.jpg")).toMatchSnapshot();
      expect(hparse("@source:dir/subdir/file.jpg")).toMatchSnapshot();
    });

    it("should return the right value without file", () => {
      expect(hparse("@source:dir")).toMatchSnapshot();
    });

    it("should return the right value without dir", () => {
      expect(hparse("@source:file.jpg")).toMatchSnapshot();
    });

    it("should return the right value without dir and file", () => {
      expect(hparse("@source:.")).toMatchSnapshot();
    });
  });

  describe("path2hfile", () => {
    it("should return the right value", () => {
      expect(path2hfile("source", "dir/file.jpg")).toMatchSnapshot();
      expect(path2hfile("source", "dir/subdir/file.jpg")).toMatchSnapshot();
    });

    it("should return the right value without dir", () => {
      expect(path2hfile("source", "file.jpg")).toMatchSnapshot();
    });

    it("should return the right value with emtpy path", () => {
      expect(path2hfile("source", "")).toMatchSnapshot();
    });
  });

  describe("hfile2path", () => {
    it("should return the right value", () => {
      expect(hfile2path(baseHfile)).toEqual("dir/file.jpg");
    });

    it("should return the right value without dir", () => {
      expect(hfile2path(omit(baseHfile, ["dir"]))).toEqual("file.jpg");
    });

    it("should return the right value without file", () => {
      expect(hfile2path(omit(baseHfile, ["file"]))).toEqual("dir");
    });

    it("should return the right value without dir and file", () => {
      expect(hfile2path(omit(baseHfile, ["dir", "file"]))).toEqual("");
    });
  });

  describe("thumbnailPath", () => {
    it("should return the right value", () => {
      expect(thumbnailPath("/root", baseHfile)).toEqual(
        "/root/.howdypix/source/dir/file.jpg"
      );
    });

    it("should return the right value without dir", () => {
      expect(thumbnailPath("/root", omit(baseHfile, ["dir"]))).toEqual(
        "/root/.howdypix/source/file.jpg"
      );
    });

    it("should return the right value without file", () => {
      expect(thumbnailPath("/root", omit(baseHfile, ["file"]))).toEqual(
        "/root/.howdypix/source/dir"
      );
    });

    it("should return the right value without file or dir", () => {
      expect(thumbnailPath("/root", omit(baseHfile, ["dir", "file"]))).toEqual(
        "/root/.howdypix/source"
      );
    });

    it("should return the right value if a HPath is passed", () => {
      expect(thumbnailPath("/root", "@source:test")).toEqual(
        "/root/.howdypix/source/test"
      );
    });
  });

  describe("splitHowdyfiles", () => {
    it("should return the right value", () => {
      expect(splitHowdyfiles(baseHfile)).toMatchSnapshot();
      expect(
        splitHowdyfiles({ ...baseHfile, dir: "dir/subdir/subsubdir" })
      ).toMatchSnapshot();
    });
  });
});
