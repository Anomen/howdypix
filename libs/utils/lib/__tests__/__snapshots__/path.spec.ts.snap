// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`path hparse should return the right value 1`] = `
Object {
  "dir": "dir",
  "file": "file.jpg",
  "source": "source",
}
`;

exports[`path hparse should return the right value 2`] = `
Object {
  "dir": "dir/subdir",
  "file": "file.jpg",
  "source": "source",
}
`;

exports[`path hparse should return the right value without dir 1`] = `
Object {
  "dir": ".",
  "file": "file.jpg",
  "source": "source",
}
`;

exports[`path hparse should return the right value without dir and file 1`] = `
Object {
  "source": "source",
}
`;

exports[`path hparse should return the right value without file 1`] = `
Object {
  "dir": "dir",
  "source": "source",
}
`;

exports[`path splitHowdyfiles should return the right value 1`] = `
Array [
  Object {
    "name": "source",
    "source": "source",
  },
  Object {
    "dir": "dir",
    "name": "dir",
    "source": "source",
  },
]
`;

exports[`path splitHowdyfiles should return the right value 2`] = `
Array [
  Object {
    "name": "source",
    "source": "source",
  },
  Object {
    "dir": "dir",
    "name": "dir",
    "source": "source",
  },
  Object {
    "dir": "dir/subdir",
    "name": "subdir",
    "source": "source",
  },
  Object {
    "dir": "dir/subdir/subsubdir",
    "name": "subsubdir",
    "source": "source",
  },
]
`;

exports[`path path2hfile should return the right value 1`] = `
Object {
  "dir": "dir",
  "file": "file.jpg",
  "source": "source",
}
`;

exports[`path path2hfile should return the right value 2`] = `
Object {
  "dir": "dir/subdir",
  "file": "file.jpg",
  "source": "source",
}
`;

exports[`path path2hfile should return the right value with emtpy path 1`] = `
Object {
  "dir": "",
  "file": "",
  "source": "source",
}
`;

exports[`path path2hfile should return the right value without dir 1`] = `
Object {
  "dir": "",
  "file": "file.jpg",
  "source": "source",
}
`;
